### ISSUE TITLE
Better handling of low-fee TXs

### ISSUE THREAD

**Issue by pinheadmz** [handshake-org/hsd#430]
Some low-fee transactions are propagating around the network somehow even though they are below the `minRelay` rate for the network.

One such TX has been "stuck" in the mempool for over a month:

https://shakescan.com/transaction/a239970210b948d462adf8a07fb2e8bf9b301ba34942cb01b86eb96aeec2ad30

```
Fee   0.000062 HNS
Time  2020-03-05 01:49
```
(Posting this issue on `2020-04-16`)

Another such TX was sent to me by a user and I wrote a script to check its standardness and fee rate: https://gist.github.com/pinheadmz/406c0d28751c969e46d4ed57823a26df

```
check undefined
stdinputs true
min fee 270
tx  fee 2
is free false
```

With default settings, nodes should allow a small rate of "free" TXs into their mempool:

https://github.com/handshake-org/hsd/blob/2d5ab277374ef751565caeaac84166c713ef9f4f/lib/mempool/mempool.js#L1576-L1595

**However, these low-fee TXs are either not getting relayed to miners, or simply not included in blocks.** Since there is no RBF in hsd, they are stuck forever (or until enough network nodes reset with clear mempools). This could also be an issue with the mining code not including low-fee TXs even if there is room in the block.

**Comment by pinheadmz** [comment]
## Solution 1

We should consider bumping up the minimum block weight value:

https://github.com/handshake-org/hsd/blob/2d5ab277374ef751565caeaac84166c713ef9f4f/lib/protocol/policy.js#L133-L141

The policy of `0` minimum weight comes from bcoin in November 2016: https://github.com/bcoin-org/bcoin/commit/bc00697adbe6ae3c4ea5960b1cd2c00279fb23a7
So that would be just after SegWit adoption and a very busy time for Bitcoin, definitely no need to allow low-fee transactions.

## Solution 2

Expose `minWeight` as a Full Node option and socially convince miners to set it to some value like at least `1000` maybe even `10000`

---

I ran test locally using `rpc settxfee 0.000001` and by adjusting these values was able to confirm low-fee TXs.

**Comment by HarukaMa** [comment]
I think with solution 1 and 2, if the chain is busy enough and the parameter is low like 1000, there will still be issues that free txs are not being mined for quite a while until there are some lucky block streaks and without RBF there will still be no way to boost it.

**Comment by randomlogin** [comment]
Hi! I've experienced this kind of problem when set too low fee by accident. And it was unpleasant surprise, there is no any mechanism for this problem. 

First of all it's unclear, why if the protocol allows low-fee txs, they are not getting mined. So I propose
## Solution 0
Find out what really causes low-fee txs to not be included in blocks.

Some thoughts on other solutions:
I'd say solutions 1-2 are just for the 'healthy' network, the low-fee tx issue solution is more a side effect. The problem they solve (blocks with too few txs) may occur in future, which may require some other actions to be done and it potentially may break these solutions. 

I like most solution 5, however it seems the hardest to implement. 

Solution 4 with time interval of say several days looks fine.

Solution 3 feels unnatural.

**Comment by pinheadmz** [comment]
closed by #514 and #515
